### ISSUE TITLE
Improve documentation on language fallbacks and on supportedLngs

### ISSUE THREAD

**Issue by svdHero** [i18next/i18next#1568]
## 🚀 Feature Proposal

While https://www.i18next.com/principles/fallback#language-fallback has some great examples, I think it is not sufficient for beginners to understand the relationship of all the parameters and moving parts involved.

In general, I have the following questions:

1. What are the actual effects and implications of option parameter `supportedLngs` on
    1. the function `i18n.changeLanguage`?
    2. the auto-detect algorithm of `languageDetector from `i18next-browser-languagedetector`?
    3. the loading of translation files in `HttpApi` from `i18next-http-backend`?
2. When I only provide translation files for `de-AT` should I also provide translation files for just `de`?
3. If not, what happens when a user whose browser locale is `de-CH` visits my page? Is there an automatical fallback from `de-CH` to `de-AT`? I.e., is there some sort of automatical fallback between different regions within the same language from any `lngA-rgnX` to `lngA-rgnY`, also compare #1452.
6. Is there some sort of automatical fallback from any region `lng-rgn` of a given language to the language `lng` itself? Or does that have to be provided manually via the `fallbackLng` parameter?

Moreover, when initializing via
```js
i18next.use(HttpApi).use(LanguageDetector).use(initReactI18next).init(options)
```
and given the following options and browser locales, what would
1. be the resulting `i18n.language`?
2. be the language/region-subfolders which `HttpApi` would expect to be present?

`navigator.languages` | `supportedLngs` | `fallbackLng` | :arrow_right: `i18n.language` | :arrow_right: necessary subfolders
---------------------- | ---------------- | ------------ | ------------------------------ | ---------------------------------
`[de-CH, de, en-US, en]` | `["de"]` | `{ "default": ["en"] }` | :question: | :question: 
`[de-CH, de, en-US, en]` | `["de-AT"]` | `{ "de-AT": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, de, en-US, en]` | `["de-AT"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, de, en-US, en]` | `["de-AT", "de"]` | `{ "de-AT": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, de, en-US, en]` | `["de-AT", "de"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, de, en-US, en]` | `["de-AT"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 
`[de-CH, de, en-US, en]` | `["de-CH"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, de, en-US, en]` | `["de-CH", "de"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, de, en-US, en]` | `["de-CH"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 
`[de-CH, de, en-US, en]` | `["de-CH", "de-AT"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 
`[de-CH, de, en-US, en]` | `["de-AT", "de-CH"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 
`[de-CH, en-US, en]` | `["de"]` | `{ "default": ["en"] }` | :question: | :question: 
`[de-CH, en-US, en]` | `["de-AT"]` | `{ "de-AT": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, en-US, en]` | `["de-AT"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, en-US, en]` | `["de-AT", "de"]` | `{ "de-AT": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, en-US, en]` | `["de-AT", "de"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, en-US, en]` | `["de-AT"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 
`[de-CH, en-US, en]` | `["de-CH"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, en-US, en]` | `["de-CH", "de"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de-CH, en-US, en]` | `["de-CH"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 
`[de-CH, en-US, en]` | `["de-CH", "de-AT"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 
`[de-CH, en-US, en]` | `["de-AT", "de-CH"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 
`[de, en-US, en]` | `["de"]` | `{ "default": ["en"] }` | :question: | :question: 
`[de, en-US, en]` | `["de-AT"]` | `{ "de-AT": ["de"], "default": ["en"] }` | :question: | :question: 
`[de, en-US, en]` | `["de-AT"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de, en-US, en]` | `["de-AT", "de"]` | `{ "de-AT": ["de"], "default": ["en"] }` | :question: | :question: 
`[de, en-US, en]` | `["de-AT", "de"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de, en-US, en]` | `["de-AT"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 
`[de, en-US, en]` | `["de-CH"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de, en-US, en]` | `["de-CH", "de"]` | `{ "de-CH": ["de"], "default": ["en"] }` | :question: | :question: 
`[de, en-US, en]` | `["de-CH"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 
`[de, en-US, en]` | `["de-CH", "de-AT"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 
`[de, en-US, en]` | `["de-AT", "de-CH"]` | `{ "de-CH": ["de-AT"], "default": ["en"] }` | :question: | :question: 

The list above is pretty long and I probably forgot quite a few combinatorical cases. However, having such a table in the docs would probably help a great deal to figure out how the fallback strategy works.
I could also be useful for unit testing the library code. Maybe such a table of uses cases already exists as test case factory?

## Motivation

Help beginners to figure out how to make use of the option parameters and reduce the number of noob github issues concerning fallbacks.

**Comment by jamuhl** [comment]
The documentation is OpenSource too and can be modified by any contributor...feel free to add a PR to the docs repo adding such details.

**Comment by svdHero** [comment]
@jamuhl I don't understand why you closed the issue. I genuinely do not know the answers to the questions above and the documentations seems to have some gaps here. If I knew how to fill these gaps, I could open a PR and help.

By closing this issue, though, you make this issue less discoverable for other users with similar questions.

Could you please reopen and help with the answers, please? I would be very grateful.

Thank you once again for this great library and all the work you're putting into it.

**Comment by jamuhl** [comment]
This is not an issue - there is no defect.

You got a question that is easily solvable by testing it out - you don't really expect me to fill your table (life is too short)

**Comment by svdHero** [comment]
Maybe I missunderstood, but I opened a feature request for documentation improvement, not a bug report. I thought this would be the proper process to report gaps in the documentation, like in other Github projects with a discussion about the missing information.

And no, I did not expect you to fill in the table, but I was hoping that some one who was involved on the implementation would add some explanation of the fallback logic to the documentation (see my questions above, not the table). After all that's what a documentation is good for, isn't? Describing how things work, so that people would know how to use them.

I don't know about the i18next project structure, it's community and how many people are involved in this. But the questions above can easily be answered by the people who implemented the funcitonality.
When it comes to the use cases in the table, I certainly have tried a few of those use cases, but between all the console warnings, it is hard to recognize a pattern. Again, this would be easy for the people who have implemented it.

To sum this up, let me say again that my intension was not to annoy you, but, instead, prevent you from having to read through future nooby questions. Improving the documentation now could help to extend your short life in the long run. :wink:

**Comment by jamuhl** [comment]
You opened a request - and I answered feel free to improve it. Like said...contributions are very welcome...

Detector -> detects one language and stores it in i18next.language
Resolver -> takes passed in language and adds non specific variant (eg. en for en-US) and fallback depending on detection
SupportedLanguages -> does not allow what the detector passes if not in the list

BackendConnector loads using plugin all the languages in the list getting from the resolver...

**Comment by svdHero** [comment]
I thought "contributions are very welcome" meant "contribute your ideas about what's missing in the docs". Beyond that I cannot contribute anything (yet), because I do not know how things are implemented. I am new to JavaScript and Front End Dev, so I do not feel confident enough to go through the source code in order to make assumptions on how things work.

Anyway, thanks again for your work and take care.
